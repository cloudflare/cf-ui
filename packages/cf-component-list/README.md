# cf-component-list

> Cloudflare List Component

## Installation

```sh
$ npm install cf-component-list
```

## Usage

```js
const React = require('react');
const {List, ListItem} = require('cf-component-list');

export default class Application extends React.Component {
  render() {
    return (
      <div>
        <h1>Cloudflare List Component Example</h1>

        <p>Basic usage:</p>
        <List>
          <ListItem>One</ListItem>
          <ListItem>Two</ListItem>
        </List>

        <p>Ordered list:</p>
        <List ordered>
          <ListItem>One</ListItem>
          <ListItem>Two</ListItem>
        </List>

        <p>Unstyled list:</p>
        <List unstyled>
          <ListItem>One</ListItem>
          <ListItem>Two</ListItem>
        </List>
      </div>
    );
  }
}
```
