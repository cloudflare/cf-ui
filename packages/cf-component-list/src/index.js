import { applyTheme } from 'cf-style-container';

import UnorderedListUnstyled from './UnorderedList';
import UnorderedListTheme from './UnorderedListTheme';
import OrderedListUnstyled from './OrderedList';
import OrderedListTheme from './OrderedListTheme';

import ListItemUnstyled from './ListItem';
import ListItemTheme from './ListItemTheme';

const UnorderedList = applyTheme(UnorderedListUnstyled, UnorderedListTheme);
const OrderedList = applyTheme(OrderedListUnstyled, OrderedListTheme);
const ListItem = applyTheme(ListItemUnstyled, ListItemTheme);

export {
  UnorderedList,
  UnorderedListUnstyled,
  UnorderedListTheme,
  OrderedList,
  OrderedListUnstyled,
  OrderedListTheme,
  ListItem,
  ListItemUnstyled,
  ListItemTheme
};
