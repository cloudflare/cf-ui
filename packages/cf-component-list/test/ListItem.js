import React from 'react';
import renderer from 'react-test-renderer';
import { ListItem } from 'cf-component-list';

test('should render', () => {
  const component = renderer.create(<ListItem>ListItem</ListItem>);
  expect(component.toJSON()).toMatchSnapshot();
});
