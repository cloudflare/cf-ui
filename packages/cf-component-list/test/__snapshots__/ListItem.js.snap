// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<li
  className=""
>
  ListItem
</li>
`;

exports[`should render 2`] = `
"

"
`;
